tests/git-scripts/pre-push.sh

```diff
@@ -32,18 +32,9 @@ echo "URL is $URL"
 
 set -eu
 
-run_test()
-{
-    TEST=$1
-    echo "running '$TEST'"
-    if ! `$TEST > /dev/null 2>&1`; then
-        echo "test '$TEST' failed"
-        return 1
-    fi
+# shorten output but be sure to catch any error
+not() {
+    ! "$@"
 }
-
-run_test ./tests/scripts/check-doxy-blocks.pl
-run_test ./tests/scripts/check-names.sh
-run_test ./tests/scripts/check-generated-files.sh
-run_test ./tests/scripts/check-files.py
-run_test ./tests/scripts/doxygen.sh
+{ tests/scripts/all.sh -k 'check_*' || echo "FAILED all.sh"; } 2>&1 |
+    not grep FAILED
```

Comment:
Why did you define a wrapper function for `!`?

Reply:
I assume because `| !` was rejected by my shell when I tried it.

Reply:
Oh right. `!` goes at the beginning of a pipeline:
```
! { … } 2>&1 | grep FAILED
```

Reply:
Marking as resolved as the entire question is avoided by the new approach of with `all.sh -q`.